src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -924,18 +924,18 @@ def __call__(
         lora_scale = (
             self.cross_attention_kwargs.get("scale", None) if self.cross_attention_kwargs is not None else None
         )
-
-        prompt_embeds, negative_prompt_embeds = self.encode_prompt(
-            prompt,
-            device,
-            num_images_per_prompt,
-            self.do_classifier_free_guidance,
-            negative_prompt,
-            prompt_embeds=prompt_embeds,
-            negative_prompt_embeds=negative_prompt_embeds,
-            lora_scale=lora_scale,
-            clip_skip=self.clip_skip,
-        )
+        if prompt_embeds is None and negative_prompt_embeds is None:
```

Comment:
this is not correct. Even if `prompt_embeds` is not None `self.encode_prompt` should be called (see how the `prompt_embeds` variable is passed to the function)

Reply:
Then please see https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py#L1239, 
when  `prompt_embeds` is given, the model does not give prompt to the pipeline.